src/main/java/com/bakdata/common_kafka_streams/CleanUpRunner.java

```diff
@@ -0,0 +1,159 @@
+package com.bakdata.common_kafka_streams;
+
+import static com.bakdata.common_kafka_streams.KafkaStreamsApplication.RESET_SLEEP_MS;
+
+import io.confluent.kafka.schemaregistry.client.CachedSchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import kafka.tools.StreamsResetter;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.kafka.clients.admin.AdminClient;
+import org.apache.kafka.streams.KafkaStreams;
+import org.apache.kafka.streams.Topology;
+import org.apache.kafka.streams.TopologyDescription.Node;
+import org.apache.kafka.streams.TopologyDescription.Processor;
+import org.apache.kafka.streams.TopologyDescription.Sink;
+
+
+@Slf4j
+public class CleanUpRunner {
+    private final Topology topology;
+    private final String streamsId;
+    private final Properties kafkaProperties;
+    private final SchemaRegistryClient client;
+    private final List<String> inputTopics;
+    private final String brokers;
+    private final boolean deleteOutputTopic;
+    private final KafkaStreams streams;
+
+    public CleanUpRunner(final KafkaStreamsApplication application) {
+        this.topology = application.createTopology();
+        this.streamsId = application.getUniqueAppId();
+        this.kafkaProperties = application.getKafkaProperties();
+        this.client = new CachedSchemaRegistryClient(application.getSchemaRegistryUrl(), 100);
+        this.inputTopics = application.getInputTopics();
+        this.brokers = application.getBrokers();
+        this.deleteOutputTopic = application.isDeleteOutputTopic();
+        this.streams = application.getStreams();
+    }
+
+    public void run() {
+        runResetter(String.join(",", this.inputTopics), this.brokers, this.streamsId);
+        if (this.deleteOutputTopic) {
+            this.deleteTopics();
+        }
+        this.streams.cleanUp();
+        try {
+            Thread.sleep(RESET_SLEEP_MS);
+        } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
+            throw new RuntimeException(e);
+        }
+    }
+
+    protected static void runResetter(final String inputTopics, final String brokers, final String appId) {
+        final String[] args = {
+                "--application-id", appId,
+                "--bootstrap-servers", brokers,
+                "--input-topics", inputTopics
+        };
+        final StreamsResetter resetter = new StreamsResetter();
+        resetter.run(args);
+    }
+
+    protected void deleteTopics() {
+        final List<Node> nodes = this.getNodes(this.topology);
```

Comment:
I would move the getNodes method etc. to another class. Also: does getNodes return input topics? We do not want to delete those in any case